crates/solidity/inputs/language/src/definition.rs

```diff
@@ -6632,448 +5998,834 @@ codegen_language_macros::compile!(Language(
+                ),
+                BuiltInType(
+                    name = "$StringType",
+                    fields = [],
+                    functions = [BuiltInFunction(
+                        name = "concat",
+                        parameters = ["string[] stringsToConcatenate"],
+                        return_type = "string memory"
+                    )]
+                ),
+                BuiltInType(
+                    name = "$TransactionType",
+                    fields = [
+                        BuiltInField(definition = "uint gasprice"),
+                        BuiltInField(
+                            definition = "address payable origin",
+                            enabled = Till("0.8.0")
+                        ),
+                        BuiltInField(definition = "address origin", enabled = From("0.8.0"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "$ContractTypeType",
+                    fields = [
+                        BuiltInField(definition = "string name"),
+                        BuiltInField(definition = "bytes creationCode", enabled = From("0.5.3")),
+                        BuiltInField(definition = "bytes runtimeCode", enabled = From("0.5.3")),
+                        BuiltInField(definition = "bytes4 interfaceId", enabled = From("0.6.7"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "$InterfaceTypeType",
+                    fields = [
+                        BuiltInField(definition = "string name"),
+                        BuiltInField(definition = "bytes4 interfaceId", enabled = From("0.6.7"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "$IntTypeType",
+                    fields = [
+                        BuiltInField(definition = "int min", enabled = From("0.6.8")),
+                        BuiltInField(definition = "int max", enabled = From("0.6.8"))
+                    ],
+                    functions = []
+                ),
+                BuiltInType(
+                    name = "$UserDefinedValueType",
+                    fields = [],
+                    functions = [
+                        BuiltInFunction(
+                            name = "wrap",
+                            parameters = ["$WrappedType elementaryType"],
+                            return_type = "$UserType"
+                        ),
+                        BuiltInFunction(
+                            name = "unwrap",
+                            parameters = ["$UserType userType"],
+                            return_type = "$WrappedType"
+                        )
+                    ],
+                    enabled = From("0.8.8")
+                ),
+                BuiltInVariable(definition = "$Function $placeholder"),
```

Comment:
not sure I understand `$Function $placeholder` here. How is this used?

Reply:
This is used to bind the `_` placeholder in function modifiers [here](https://github.com/manastech/slang/blob/built-in-contexts/crates/solidity/inputs/language/bindings/rules.msgb#L1618-L1623). But... I think we could get away with defining it as `BuiltInVariable(definition = "$Function _")` and avoid the "translation". We couldn't before because we needed to parse a Solidity file with this definition.

And come to think of it, we can now replace all `$` in built-ins definitions directly with `%`.

Reply:
Thanks for explaining. I suggest adding a comment to explain this:

```rs
                // Placeholder for modifiers, to be used to inject the modified function body (`_;`)
                BuiltInVariable(definition = "%Function _"),
```